scripts/new-chain.ts

```diff
@@ -0,0 +1,234 @@
+			labelOperation: "--remove-label"
+		}
+	} finally {
+		const summaryFile = process.env.SUMMARY_FILE
+		if (summaryFile) {
+			fs.writeFileSync(summaryFile, JSON.stringify(summary, null, 2))
+		} else {
+			console.log(summary)
+		}
+	}
+}
+
+async function verifyNewChainRequest() {
+	// Extract the RPC URL (first URL) from the issue body as a string
+	const rpcUrl = process.env.RPC
+	if (!rpcUrl) {
+		throw NewChainError.rpcNotFound()
+	}
+
+	const provider = new ethers.providers.JsonRpcProvider(rpcUrl)
+	const { chainId } = await provider.getNetwork()
+	console.log({ chainId })
+	const filePath = path.join(__dirname, "..", "artifacts", `${chainId}`, "deployment.json")
+
+	// Check if the factory is already deployed
+	// If the file exists, it means the factory is already deployed
+	// If the file does not exist, it means the factory is not deployed or not added to the repository
+	// and we can proceed with the deployment
+	const deployed = fs.existsSync(filePath)
+	if (deployed) {
+		throw NewChainError.factoryAlreadyDeployed()
+	}
+
+	// Check if the chain is listed in the chainlist
+	// If the chain is listed, we can proceed with the deployment
+	const chainlist = `https://raw.githubusercontent.com/ethereum-lists/chains/master/_data/chains/eip155-${chainId}.json`
+	const { ok: onChainlist } = await fetch(chainlist)
+	console.log({ chainlist, onChainlist })
+	if (!onChainlist) {
+		throw NewChainError.chainNotListed(chainId.toString())
+	}
+
+	const nonce = await provider.getTransactionCount(SIGNER)
+	const code = await provider.getCode(ADDRESS)
+	const codehash = ethers.utils.keccak256(code)
+	console.log({ nonce, codehash, code })
+	// Check if any code is deployed at the address
+	if (ethers.utils.hexDataLength(code) > 0) {
+		// Check if the codehash matches the expected codehash
+		if (codehash !== CODEHASH) {
+			throw NewChainError.factoryDifferentBytecode()
+		}
+
+		if (nonce === 0) {
+			throw NewChainError.factoryPreDeployed()
+		} else {
+			throw NewChainError.factoryNotAddedToRepo()
+		}
+		// TODO: Create a PR to add the artifact to the repository
+	} else if (nonce > 0) {
+		throw NewChainError.factoryDeployerAccountNonceBurned()
+	} else {
+		// Get the gas price and gas limit
+		const gasPrice = await provider.getGasPrice()
+		if(!gasPrice) {
+			throw NewChainError.gasPriceNotRetrieved()
+		}
+		let gasLimit;
+		try {
+			gasLimit = await provider.estimateGas({
+				from: SIGNER,
+				data: FACTORY_BYTECODE,
+			})
+			if(!gasLimit) {
+				throw NewChainError.gasLimitNotEstimated()
+			}
+		} catch (error) {
+			throw NewChainError.gasLimitEstimationFailed()
+		}
+
+		const gasEstimate = gasPrice.mul(gasLimit!).mul(15).div(10) // 15% buffer
+		console.log({ gasPrice: gasPrice.toString(), gasLimit: gasLimit?.toString(), gasEstimate: gasEstimate.toString() })
+
+		// Get the deployed bytecode simulation
+		let simulation: string = '';
```

Comment:
TS should let you do this:

```suggestion
		let simulation: string;
```

Reply:
Updated 👍🏾